packages/mui-styles/test/theme-scoping.test.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import { expect } from 'chai';
+import { createRenderer } from 'test/utils';
+import * as material from '@mui/material/styles';
+import * as joy from '@mui/joy/styles';
+import { makeStyles } from '@mui/styles';
+
+describe('Theme scoping', () => {
+  const { render } = createRenderer();
+
+  it('works without theme scoping', () => {
+    const useStyles = makeStyles<material.Theme>((theme) => ({
+      root: {
+        color: theme.palette.primary.main,
+      },
+    }));
+    function Component() {
+      const classes = useStyles();
+
+      return <div className={classes.root}>Component</div>;
+    }
+
+    expect(() =>
+      render(
+        <material.ThemeProvider theme={material.createTheme()}>
+          <Component />
+        </material.ThemeProvider>,
+      ),
+    ).not.to.throw();
+  });
+
+  it('theme scoping works', () => {
+    const useStyles = makeStyles<material.Theme>((theme) => ({
+      root: {
+        color: theme.palette.primary.main,
+      },
+    }));
+    function Component() {
+      const classes = useStyles();
+
+      return <div className={classes.root}>Component</div>;
+    }
+
+    expect(() =>
+      render(
+        <material.ThemeProvider theme={{ [material.THEME_ID]: material.createTheme() }}>
+          <Component />
+        </material.ThemeProvider>,
+      ),
+    ).not.to.throw();
+  });
+
+  it('should get Material UI theme even it is inside Joy provider', () => {
+    const useStyles = makeStyles<material.Theme>((theme) => ({
+      root: {
+        color: theme.palette.grey[700], // joy does not have `grey` in the default theme.
+      },
+    }));
+    function Component() {
+      const classes = useStyles();
+
+      return <div className={classes.root}>Component</div>;
+    }
+
+    expect(() =>
+      render(
+        <material.ThemeProvider theme={{ [material.THEME_ID]: material.createTheme() }}>
+          <joy.ThemeProvider>
+            <Component />
+          </joy.ThemeProvider>
+        </material.ThemeProvider>,
+      ),
+    ).not.to.throw();
```

Comment:
We are again just checking that it doesn't throw, not ensuring the right color is used.

Reply:
Added a test for that.